others/freivalds_algorithm.cpp

```diff
@@ -0,0 +1,169 @@
+/**
+ * @file
+ * @brief Implementation of [Freivalds' algorithm]
+ * (https://en.wikipedia.org/wiki/Freivalds%27_algorithm)
+ * @author [Darshana Sarma](https://github.com/Darshana-Sarma)
+ * @details
+ * Given three n × n matrices A, B, and
+ * C, a general problem is to verify whether AB=C
+ *
+ * The time Complexity of the algorithm is O(n^{2}).
+ */
+#include <array>     //   for std::array
+#include <cassert>   //   for assert
+#include <iostream>  //   for io operations
+
+/**
+ * @namespace others
+ * @brief Other algorithms
+ */
+namespace others {
+/**
+ * @namespace freivalds
+ * @brief Functions for [Freivalds']
+ * (https://en.wikipedia.org/wiki/Freivalds%27_algorithm) algorithm
+ */
+namespace freivalds {
+/**
+ * The function will check AB=C by applying Freidvald's algorithm 
+ * @tparam N dimension of square matrix NxN
+ * @param A square matrix A
+ * @param B square matrix B
+ * @param C resultant square matrix C
+ * @returns boolean value
+ */
+template <std::size_t N>
+bool CoppersmithAlgo(std::array<std::array<double, N>, N> A,

```

Comment:
pass the input parameters by const reference
```suggestion
bool CoppersmithAlgo(const std::array<std::array<double, N>, N>& A,
```